Backports for 1.12.0-alpha1

KristofferC opened the pull request:
Backported PRs:
- [x] #57142 <!-- Add reference to time_ns in time -->
- [x] #57241 <!-- Handle `waitpid` race condition when `SIGCHLD` is set to `SIG_IGN` -->
- [x] #57249 <!-- restore non-freebsd-unix fix for profiling -->
- [x] #57211 <!-- Ensure read/readavailable for BufferStream are threadsafe -->
- [x] #57262 <!-- edit NEWS for v1.12 -->
- [x] #57226 <!-- cfunction: reimplement, as originally planned, for reliable performance -->
- [x] #57253 <!-- bpart: Fully switch to partitioned semantics -->
- [x] #57273 <!-- fix "Right arrow autocompletes at line end" implementation -->
- [x] #57280 <!-- dep: Update JuliaSyntax -->
- [x] #57229 <!-- staticdata: Close data race after backedge insertion -->
- [x] #57298 <!-- Updating binding version to fix MMTk CI -->
- [x] #57248 <!-- improve concurrency safety for `Compiler.finish!` -->
- [x] #57312 <!-- Profile.print: de-focus sleeping frames as gray -->
- [x] #57289 <!-- Make `OncePerX` subtype `Function` -->
- [x] #57310 <!-- Make ptls allocations at least 128 byte aligned -->
- [x] #57311 <!-- Add a warning for auto-import of types -->
- [x] #57338 <!-- fix typo in Float32 random number generation -->
- [x] #57293 <!-- Fix getfield_tfunc when order or boundscheck is Vararg -->
- [x] #57349 <!-- docs: fix-up world-age handling for META access -->
- [x] #57344 <!-- Add missing type asserts when taking the queue out of the task struct -->
- [x] #57348 <!-- 🤖 [master] Bump the SparseArrays stdlib from 212981b to 72c7cac -->
- [x] #55040 <!-- Allow macrocall as function sig -->
- [x] #57299 <!-- Add missing latestworld after parameterized type alias -->

Non-merged PRs with backport label:
- [ ] #57342 <!-- trimming: make sure to fail / warn on `Expr(:call, ...)` -->
- [ ] #57317 <!-- internals: add _defaultctor function for defining ctors -->
- [ ] #57275 <!-- Compiler: fix unsoundness of getfield_tfunc on Tuple Types -->
- [ ] #57274 <!-- udpate path to Compiler stdlib to a valid one similar to how it is done for other stdlibs -->
- [ ] #57270 <!-- 🤖 [master] Bump the LinearAlgebra stdlib from e7da19f to 924dda4 -->
- [ ] #57266 <!-- 🤖 [master] Bump the Statistics stdlib from d49c2bf to 77bd570 -->
- [ ] #57143 <!-- Add JuliaLang/JuliaSyntax.jl#525 to NEWS.md, flisp parser, and REPL -->
- [ ] #55853 <!-- don't mutate globals when constructing `Rational` from `AbstractIrrational` -->
- [ ] #49933 <!-- Allow for :foreigncall to transition to GC safe automatically -->

nlw0 committed:
Add reference to time_ns in time (#57142)

I couldn't find `time_ns` when I was looking for it, nice to make clear
the "monotonic" clock is also available in Base.

(cherry picked from commit a52de835af96c2f5f3939fa3342d985e613f5a30)

topolarity committed:
Handle `waitpid` race condition when `SIGCHLD` is set to `SIG_IGN` (#57241)

The `fork()` we do here relies on `SIGCHLD` to make sure that we don't
race against the child.

This is easy to see in an embedding application that dynamically links
`libjulia`:
```c
int main(int argc, char *argv[])
{
    signal(SIGCHLD, SIG_IGN);
    void *handle = dlopen("path/to/libjulia.so", RTLD_LAZY);
    return 0;
}
```

Without this change, this fails with an error message:
```
Error during libstdcxxprobe in parent process:
waitpid: No child processes
```

Resolves #57240

(cherry picked from commit daf865e8605344764c1f26b5fc4412f100888791)

DilumAluthge commented:
@KristofferC Should this PR target `release-1.12` (instead of `master`)?

KristofferC commented:
Oops, yes, thanks!

DilumAluthge commented:
I think I'll have to make a commit to this branch just for the sake of kicking off 1.12 Buildkite for the first time - I'll immediately revert the commit, and from then on CI should work correctly.

DilumAluthge commented:
The next time you force-push to this branch, you can drop both https://github.com/JuliaLang/julia/pull/57258/commits/f15d417a24c2cf720d4cef73c83e2b52d5be9f33 and https://github.com/JuliaLang/julia/pull/57258/commits/8bb55bd3916bccf9bbe19ec8920d1bd3171c606a.

DilumAluthge commented:
Looks like the 1.12 Buildkite CI has now been triggered correctly on this PR.

vtjnash committed:
restore non-freebsd-unix fix for profiling (#57249)

Restores #57035, undo #57089 for non-FreeBSD. While I suggested doing
this change for all platforms, I forgot that means non-FreeBSD platforms
become vulnerable again to the very deadlock problems that #57035 was
required to prevent. That fix seems to not be viable on FreeBSD due to
known libc implementation problems on that platform. However, upon
closer inspection of the questionable design implementation decisions
they seem to have made here, the platform is likely not currently
vulnerable to this libunwind bug in the first place:
https://github.com/lattera/freebsd/blob/master/libexec/rtld-elf/rtld_lock.c#L120

(cherry picked from commit 2f0a5239f00697914128e7bb7b58f7dadec0d519)

quinnj committed:
Ensure read/readavailable for BufferStream are threadsafe (#57211)

It looks like these methods were just missed while overloading for
BufferStream.

There's also `readbytes!` where the current implementation will fallback
to the `LibuvStream` implementation that is currently not threadsafe.
What's the best approach there since the implementation is quite a bit
more involved? Just duplicate the code but for BufferStream? Should we
take the BufferStream lock and invoke the LibuvStream method? Open to
ideas there.

Also open to suggestions for having tests here? Not easy to simulate the
data race of writing and calling readavailable.

The fix here will unblock https://github.com/JuliaWeb/HTTP.jl/pull/1213
(I'll probably do some compat shim there until this is fully released).

Thanks to @oscardssmith for rubber ducking this issue with me.

Probably most helpfully reviewed by @vtjnash.

---------

Co-authored-by: Jameson Nash <vtjnash@gmail.com>
(cherry picked from commit ffc96bc7e14a80b22f3a7bbfd4fc60adaa120f1a)

DilumAluthge commented:
Alright, looks like the 1.12 Buildkite pipeline is working.

JeffBezanson committed:
edit NEWS for v1.12 (#57262)

(cherry picked from commit 99fd5d9a92190e826bc462d5739e7be948a3bf44)

vtjnash committed:
cfunction: reimplement, as originally planned, for reliable performance (#57226)

(cherry picked from commit ca7cf30d6310469c335e7d2ed95c786cf8b93881)

Keno committed:
bpart: Fully switch to partitioned semantics (#57253)

This is the final PR in the binding partitions series (modulo bugs and
tweaks), i.e. it closes #54654 and thus closes #40399, which was the
original design sketch.

This thus activates the full designed semantics for binding partitions,
in particular allowing safe replacement of const bindings. It in
particular allows struct redefinitions. This thus closes
timholy/Revise.jl#18 and also closes #38584.

The biggest semantic change here is probably that this gets rid of the
notion of "resolvedness" of a binding. Previously, a lot of the behavior
of our implementation depended on when bindings were "resolved", which
could happen at basically an arbitrary point (in the compiler, in REPL
completion, in a different thread), making a lot of the semantics around
bindings ill- or at least implementation-defined. There are several
related issues in the bugtracker, so this closes #14055 closes #44604
closes #46354 closes #30277

It is also the last step to close #24569.
It also supports bindings for undef->defined transitions and thus closes
#53958 closes #54733 - however, this is not activated yet for
performance reasons and may need some further optimization.

Since resolvedness no longer exists, we need to replace it with some
hopefully more well-defined semantics. I will describe the semantics
below, but before I do I will make two notes:

1. There are a number of cases where these semantics will behave
slightly differently than the old semantics absent some other task going
around resolving random bindings.
2. The new behavior (except for the replacement stuff) was generally
permissible under the old semantics if the bindings happened to be
resolved at the right time.

With all that said, there are essentially three "strengths" of bindings:

1. Implicit Bindings: Anything implicitly obtained from `using Mod`, "no
binding", plus slightly more exotic corner cases around conflicts

2. Weakly declared bindings: Declared using `global sym` and nothing
else

3. Strongly declared bindings: Declared using `global sym::T`, `const
sym=val`, `import Mod: sym`, `using Mod: sym` or as an implicit strong
global declaration in `sym=val`, where `sym` is known to be global
(either by being at toplevle or as `global sym=val` inside a function).

In general, you always allowed to syntactically replace a weaker binding
by a stronger one (although the runtime permits arbitrary binding
deletion now, this is just a syntactic constraint to catch errors).
Second, any implicit binding can be replaced by other implicit bindings
as the result of changing the `using`'ed module. And lastly, any
constants may be replaced by any other constants (irrespective of type).

We do not currently allow replacing globals, but may consider changing
that in 1.13.

This is mostly how things used to work, as well in the absence of any
stray external binding resolutions. The most prominent difference is
probably this one:

```
set_foo!() = global foo = 1
```

In the above terminology, this now always declares a "strongly declared
binding", whereas before it declared a "weakly declared binding" that
would become strongly declared on first write to the global (unless of
course somebody had created a different strongly declared global in the
meantime). To see the difference, this is now disallowed:

```
julia> set_foo!() = global foo = 1
set_foo! (generic function with 1 method)

julia> const foo = 1
ERROR: cannot declare Main.foo constant; it was already declared global
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
```

Before it would depend on the order of binding resolution (although it
just crashes on current master for some reason - whoops, probably my
fault).

Another major change is the ambiguousness of imports. In:
```
module M1; export x; x=1; end
module M2; export x; x=2; end
using .M1, .M2
```
the binding `Main.x` is now always ambiguous and will throw on access.
Before which binding you get, would depend on resolution order. To
choose one, use an explicit import (which was the behavior you would
previously get if neither binding was resolved before both imports).

(cherry picked from commit 888cf0350668a677ef27c87ab80a8562f3bc211e)

KristofferC committed:
fix "Right arrow autocompletes at line end" implementation (#57273)

Fixes https://github.com/JuliaLang/julia/issues/56864.

Ref
https://github.com/JuliaLang/julia/blob/99fd5d9a92190e826bc462d5739e7be948a3bf44/stdlib/REPL/src/LineEdit.jl#L504

---------

Co-authored-by: Ian Butterworth <i.r.butterworth@gmail.com>
(cherry picked from commit f0446c624b2af960c597e22a39131ced9cfce6fb)

nanosoldier commented:
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/57258#issuecomment-2634227314) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/cdf2f46_vs_d63aded/report.html) is available.

<details><summary>Report summary</summary>

## ❗ Packages that crashed

**168 packages crashed only on the current version.**

 - The process was aborted: 22 packages
 - Invalid LLVM IR was generated: 1 packages
 - An internal error was encountered: 129 packages
 - An unreachable instruction was executed: 2 packages
 - GC corruption was detected: 1 packages
 - A segmentation fault happened: 13 packages

4 packages crashed on the previous version too.

## ✖ Packages that failed

**1421 packages failed only on the current version.**

 - Package has syntax issues: 439 packages
 - Package fails to precompile: 460 packages
 - Illegal method overwrites during precompilation: 18 packages
 - Package has test failures: 76 packages
 - Package tests unexpectedly errored: 198 packages
 - Networking-related issues were detected: 2 packages
 - There were unidentified errors: 6 packages
 - Tests became inactive: 5 packages
 - Test duration exceeded the time limit: 212 packages
 - Test log exceeded the size limit: 5 packages

2709 packages failed on the previous version too.

## ✔ Packages that passed tests

**36 packages passed tests only on the current version.**

 - Other: 36 packages

4580 packages passed tests on the previous version too.

## ➖ Packages that were skipped altogether

**61 packages were skipped only on the current version.**

 - Package could not be installed: 61 packages

1300 packages were skipped on the previous version too.



</details>

inkydragon committed:
dep: Update JuliaSyntax (#57280)

- Update `JuliaSyntax.jl` to v1.0.1
- Revert workaround changes, use original test case `n37134`

Fix #57223

(cherry picked from commit 97c920dc4e12f7c6c65e681adb3f21d6f7abebdf)

Keno committed:
staticdata: Close data race after backedge insertion (#57229)

Addresses review comment in
https://github.com/JuliaLang/julia/pull/57212#discussion_r1937694448.
The key is that the hand-off of responsibility for verification between
the loading code and the ordinary backedge mechanism happens under the
world counter lock to ensure synchronization.

(cherry picked from commit 34aceb528d26c2a3529d1afe5aed3bc039fdb181)

Keno committed:
Allow macrocall as function sig (#55040)

@KristofferC requested that `function @main(args) end` should work. This
is currently a parse error. This PR makes it work as expected by
allowing macrocall as a valid signature in function (needs to exapand to
a call expr). Note that this is only the flisp changes. If this PR is
accepted, an equivalent change would need to be made in JuliaSyntax.

(cherry picked from commit b65f0040dd367f2da76f6ed4b01924a368fe50b8)

Keno committed:
Add missing latestworld after parameterized type alias (#57299)

Fixes #57267

(cherry picked from commit e553fc8c6f794118a5dfcb0506fe3a0a0cd34ee4)

nanosoldier commented:
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/57258#issuecomment-2640270237) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/7825364_vs_d63aded/report.html) is available.

<details><summary>Report summary</summary>

## ❗ Packages that crashed

**92 packages crashed only on the current version.**

 - The process was aborted: 14 packages
 - Invalid LLVM IR was generated: 1 packages
 - An internal error was encountered: 69 packages
 - An unreachable instruction was executed: 1 packages
 - GC corruption was detected: 1 packages
 - A segmentation fault happened: 6 packages


## ✖ Packages that failed

**1032 packages failed only on the current version.**

 - Package has syntax issues: 11 packages
 - Package fails to precompile: 505 packages
 - Illegal method overwrites during precompilation: 54 packages
 - Package has test failures: 75 packages
 - Package tests unexpectedly errored: 204 packages
 - Networking-related issues were detected: 1 packages
 - There were unidentified errors: 6 packages
 - Tests became inactive: 3 packages
 - Test duration exceeded the time limit: 170 packages
 - Test log exceeded the size limit: 3 packages

81 packages failed on the previous version too.

## ✔ Packages that passed tests

292 packages passed tests on the previous version too.



</details>

udesou committed:
Updating binding version to fix MMTk CI (#57298)

Updating mmtk-julia version to include
https://github.com/mmtk/mmtk-julia/pull/228 and fix the MMTk CI.

I've also changed the allocation profiler tests to skip all tests
instead of just a few since I've seen some spurious errors - they should
all be related though, we need to make sure the profiler accounts for
fastpath allocation (see
https://github.com/JuliaLang/julia/issues/57103)

This should fix https://github.com/JuliaLang/julia/issues/57306.

(cherry picked from commit 72f8a106c357de6819e07352e88e49148c8c9546)

aviatesk committed:
improve concurrency safety for `Compiler.finish!` (#57248)

Similar to JuliaLang/julia#57229, this commit ensures that
`Compiler.finish!` properly synchronizes the operations to set
`max_world` for cached `CodeInstance`s by holding the world counter
lock. Previously, `Compiler.finish!` relied on a narrow timing window to
avoid race conditions, which was not a robust approach in a concurrent
execution environment.

This change ensures that `Compiler.finish!` holds the appropriate lock
(via `jl_promote_ci_to_current`).

(cherry picked from commit 4ebb50b6a0445b6ecdd74374c3999da5b3f33b9d)

IanButterworth committed:
Profile.print: de-focus sleeping frames as gray (#57312)

(cherry picked from commit 132057c378af992f768e5b2a9b69543c0017c0d9)

christiangnrd committed:
Make `OncePerX` subtype `Function` (#57289)

(cherry picked from commit dbd528025166f92bf0041471b2ddea3027e106e6)

gbaraldi committed:
Make ptls allocations at least 128 byte aligned (#57310)

Fixes
https://github.com/JuliaLang/julia/issues/54560#issuecomment-2439710448

(cherry picked from commit 79e98e3b89da5e57f829ccde21b576be9fc8ea1b)

maleadt commented:
`CpuId`-related failures (`llvmcall` getting interpreted) bisected and filed as https://github.com/JuliaLang/julia/issues/57318

Keno committed:
Add a warning for auto-import of types (#57311)

This adds a warning for the auto-import of types cases (#25744) that we
have long considered a bit of a bug, but didn't want to change because
it is too breaking.

The reason to do it now is that the binding rework has made this case
more problematic (see #57290). To summarize, the question is what
happens when the compiler sees `f(x) = ...` and `f` is currently and
implicitly imported binding. There are two options:

1. We add a method to the generic function referred to by `f`, or
2. We create a new generic function `f` in the current module.

Historically, case 1 has the additional complication that this error'd
unless `f` is a type. It is my impression that a lot of existing code
did not have a particularly good understanding of the resolved-ness
dependence of this behavior.

However, because case 1 errors for generic functions, it appears that
existing code generally expects case 2. On the other hand, for types,
there is existing code in both directions (#57290 is an example of case
2; see #57302 for examples of case 1). That said, case 1 is more common
(because types tend to be resolved because they're used in signatures at
toplevel).

Thus, to retain compatibility, the current behavior on master (where
resolvedness is no longer available) is that we always choose case 2 for
functions and case 1 for types. This inconsistency is unfortunate, but I
tried resolving this in either way (making all situations case 1 or all
case 2) and the result was too breaking.

Nevertheless, it is problematic that there is existing code that expects
case 2 beavior for types and we should help users to know what the
correct way to fix it is. The proposed resolution is thus:
1. Retain case 1 behavior for types
2. Make it a warning to use, encouraging people to explicitly import,
since we generally consider the #25744 case a bug.

Example:
```
julia> module Foo
         String(i::Int) = i
       end
WARNING: Type Core.String was auto-`import`ed in `Foo`.
NOTE: This behavior is deprecated and may change in future Julia versions.
NOTE: This behavior may have differed in Julia versions prior to 1.12 depending on binding resolution.
Hint: To retain the current behavior, add an explicit `import Core: String` in Foo.
Hint: To create a new generic function of the same name use `function String end`.
Main.Foo
```

(cherry picked from commit 8c62f42947ee01fb17d1baf8c5e953fb3516b31f)

KristofferC committed:
fix typo in Float32 random number generation (#57338)

introduced in https://github.com/JuliaLang/julia/pull/56144

---------

Co-authored-by: Mosè Giordano <765740+giordano@users.noreply.github.com>
(cherry picked from commit dd1387854540977a77725821c2ce0249bde1e443)

xal-0 committed:
Fix getfield_tfunc when order or boundscheck is Vararg (#57293)

Even if T has no intersection with the type we want, we don't know that
we will throw because the arguments are optional.

Fixes #57292.

(cherry picked from commit 53431309e20eb057709a4f0d4de5663f4a233afd)

topolarity committed:
docs: fix-up world-age handling for META access (#57349)

This should be enough to get rid of the 3 lingering warnings in our main
test suite.

(cherry picked from commit 751a0d728e32932a3f6f57065e9e792ce9325a8c)

gbaraldi committed:
Add missing type asserts when taking the queue out of the task struct (#57344)

This was causing some dynamic dispatches (caught by the juliac test)

(cherry picked from commit 90662b669e6b0a25f8b276fe60efba2a8018711c)

DilumAluthgeBot committed:
🤖 [master] Bump the SparseArrays stdlib from 212981b to 72c7cac (#57348)

Stdlib: SparseArrays
URL: https://github.com/JuliaSparse/SparseArrays.jl.git
Stdlib branch: main
Julia branch: master
Old commit: 212981b
New commit: 72c7cac
Julia version: 1.13.0-DEV
SparseArrays version: 1.12.0(Does not match)
Bump invoked by: @ViralBShah
Powered by:
[BumpStdlibs.jl](https://github.com/JuliaLang/BumpStdlibs.jl)

Diff:
https://github.com/JuliaSparse/SparseArrays.jl/compare/212981bf29b03ba460d3251ee9aa4399931b3f2d...72c7cac6bbf21367a3c2fbc5c50e908aea5984bb

```
$ git log --oneline 212981b..72c7cac
72c7cac Explicitly declare type constructor imports (#598)
ff083ce README: update github action badge (#596)
```

Co-authored-by: ViralBShah <744411+ViralBShah@users.noreply.github.com>
(cherry picked from commit ac79b9f413784be185db0487e785c526c57d7daa)

JeffBezanson committed:
replace `Linking.__init__` with `OncePerProcess` (#57351)

This will cause these values to be populated at slightly different times
than now, but I don't think it will matter.

(cherry picked from commit 55d5a4be999cab0ee377a94698ebaa5d4929f53e)

topolarity committed:
trimming: make sure to fail / warn on `Expr(:call, ...)` (#57342)

This dispatch should not be treated as resolved just because arg0 is
constant. With the previous code, that meant it was eligible for
last-minute call resolution, but call-resolution in codegen is now
forbidden so this needs to fail unilaterally.

(cherry picked from commit 0b74d1752e4a02f9078e95bc9893619b2e52a0b6)

KristofferC committed:
add a missing type assert to `OncePerTask` callable (#57356)

also test that the difference `OncePerX` infer

(cherry picked from commit 73506ed5e7f6f40028d4851c99ec45b5dc452adc)

giordano committed:
[OpenSSL_jll] Update to v3.0.16 (#57363)

https://github.com/openssl/openssl/releases/tag/openssl-3.0.16
(cherry picked from commit 7ec984a738630d11ea976da5409553eafc4e75bc)

KristofferC committed:
use an empty symbol instead of module name as placeholder for module `file` metadata (#57359)

Based on the discussion in
https://github.com/JuliaLang/julia/pull/55963#discussion_r1786421289

(cherry picked from commit c46b5334547293fdc93fb9b5116d061bb84e9d0c)

topolarity committed:
Bump Documenter to v1.8.1 (#57362)

This should get fix the binding world-age warnings in our `doctest` CI

(cherry picked from commit 796a849d11e5a5b72df327e005dac6ff84307b65)

KristofferC committed:
update path to Compiler stdlib to a valid one similar to how it is done for other stdlibs (#57274)

Fixes #56865. The Compiler "stdlib" resides in a different location so
the current logic didn't handle updating it.

(cherry picked from commit 9339dea0426c5416e97b9a4bf2a2b52bb678ad91)

stevengj committed:
cconvert(Ref{BigFloat}, x) should return BigFloatData (#57367)

#55906 changed `cconvert(Ref{BigFloat}, x::BigFloat)` to return `x.d`,
but neglected to do so for other types of `x`, where it still returns a
`Ref{BigFloat}` and hence is now returning the wrong type for `ccall`.

Not only does this break backwards compatibility
(https://github.com/JuliaMath/SpecialFunctions.jl/issues/485), but it
also seems simply wrong: the *whole* job of `cconvert` is to convert
objects to the correct type for use with `ccall`. This PR does so (at
least for `Number` and `Ref{BigFloat}`).

(cherry picked from commit 6dca4f4eda08df9cff9626dba8cb9539818b28f8)

KristofferC commented:
@nanosoldier `runtests(["SIMDTypes", "SimpleLooper", "LazyBroadcast", "RiemannComplexNumbers", "DontMaterialize", "Fuzzy", "MistyClosures", "SyntaxTree", "TypeTree", "TraceFuns", "ProtoStructs", "Bits", "ExprTools", "FindDefinition", "MultiThreadedCaches", "CoverageTools", "FuzzyCompletions", "WidthLimitedIO", "Requires", "CountFlops", "CellLists", "MappedArrays", "MallocArrays", "PtrArrays", "Syslogs", "CatViews", "Recyclers", "StringAlgorithms", "CpuId", "SingleFloats", "DirectedAcyclicGraphs", "InterProcessCommunication", "BasesAndSamples", "CompTime", "OptimalSortingNetworks", "InternedStrings", "RoundingEmulator", "NarrativeTest", "OperatorScaling", "CircuitModelZoo", "PrecompileTools", "StarAlgebras", "Adapt", "RNGPool", "ReferenceImplementations", "CountdownNumbers", "FieldFlags", "Results", "Stuffing", "Match", "EnzymeCore", "BSON", "StringManipulation", "Interfaces", "ThreadLocalCounters", "RuntimeGeneratedFunctions", "ChangePrecision", "NNParamsPrinter", "YaoHIR", "BracedErrors", "FlatBuffers", "TypeStability", "Losers", "LambertW", "FileCmp", "IterativeRefinement", "CSTParser", "Umlaut", "WiltonInts84", "LoweredCodeUtils", "JACC", "KeywordCalls", "Jive", "LoopFieldCalc", "TypedSyntax", "ResumableFunctions", "DotCall", "BorrowChecker", "GridMaps", "TPSAInterface", "Spec", "DRIPs", "Hygienic", "Abaco", "Quadmath", "COESA", "Chevrons", "CPUSummary", "Baobzi", "H3", "SortedVectors", "YAArguParser", "LMDB", "CircularArrayBuffers", "MicroFloatingPoints", "GAFramework", "ProgressMeter", "BibInternal", "DataScienceTraits", "TrixiBase", "MaybeInplace", "Colors", "SimpleExpressions", "ConstraintCommons", "FactorLoadingMatrices", "Tracy", "StaticArraysBlasInterfaces", "RBNF", "ForwardMethods", "RollingWindowArrays", "PlutoHooks", "PiecewiseLinearFunctions", "Functors", "ArrayInterface", "GenericSchur", "BioAtomsCount", "JuliaWorkspaces", "BioVossEncoder", "OpenQASM", "FixedPointDecimals", "DimensionfulAngles", "SimJulia", "Mueller", "BitBasis", "AStarGridSearch", "HypertextTemplates", "PDMats", "FileIO", "MultiBroadcastFusion", "Accessors", "ConcurrentSim", "DispatchDoctor", "StaticLint", "GeometryTypes", "QuanticsGrids", "BioMarkovChains", "DiscretePIDs", "CAP", "CallableExpressions", "LaTeXEscapes", "CollisionDetection", "ADTypes", "SimpleI18n", "UnitfulChainRules", "AdaptiveSparseGrids", "Delaunator", "GraphQLGen", "GCMAES", "UserConfig", "GeneticProgramming", "Divergences", "MarchingCubes", "SHTns", "LaTeXTabulars", "SpinAdaptedSecondQuantization", "FastPower", "InteractiveErrors", "MCP2221Driver", "Binning2D", "ConvexHulls2d", "JuliaInterpreter", "Radiant", "GenericLinearAlgebra", "ControllerFormats", "CannotWaitForTheseOptimisers", "PolarizedTypes", "ExplicitImports", "LinearOperators", "PositionalEmbeddings", "XDiag", "FastCholesky", "GitCommand", "Malt", "BlockMatching", "SmallCollections", "DifferentiableFlatten", "SimpleCaching", "DiskBackedDicts", "SnapshotTests", "ColPack", "FCSFiles", "ExponentialAction", "MarkovKernels", "QuadGK", "DocSeeker", "SMCExamples", "SpectralKit", "Cthulhu", "RustFFT", "FiniteDifferences", "TiledIteration", "SigmaProofs", "YaoBase", "HMatrices", "ArrayLayouts", "Metatheory", "NLSolvers", "RRTMGP", "WGSLTypes", "JLLPrefixes", "UnitfulAssets", "ShuffleProofs", "CellArrays", "OpenSSLGroups", "UCX", "BoundedDegreeGraphs", "CryptoGroups", "Cosmology", "ParameterSchedulers", "DynamicQuantumCircuits", "InPartS", "Stencils", "ParameterHandling", "SquashFS", "TerminalGat", "InteratomicPotentials", "AllocCheck", "TypedMatrices", "AtomicSymmetries", "Oracle", "NearestNeighborDescent", "StructArrays", "Measurements", "QuasiArrays", "MieScattering", "Polyester", "DiffImageRotation", "FastBroadcast", "VisualGeometryDatasets", "SubSIt", "WannierIO", "NestedGraphs", "PlayingCards", "OpticalPropagation", "ADOLC", "SpecialFunctions", "PrimitiveOneHot", "AutoDiffOperators", "SIMDMathFunctions", "CounterfactualRegret", "LatinHypercubeSampling", "LabelledGraphs", "AbstractDifferentiation", "MolecularMinimumDistances", "Roots", "QXZoo", "AdmittanceModels", "ManifestUtilities", "ThreadedDenseSparseMul", "DataToolkitBase", "FiniteHorizonGramians", "Uncertain", "StableHashTraits", "GridWorlds", "Krylov", "ClimaUtilities", "Rotations", "InterferometricModels", "LFRBenchmarkGraphs", "AllocArrays", "NeidArchive", "Term", "PharmaceuticalClassification", "RxCiphers", "Proj", "MathematicalSystems", "BinaryBuilderSources", "AccurateArithmetic", "LinkedInAPI", "ImageShow", "DistanceTransforms", "ExaTron", "Groups", "MetaGraphsNext", "Overpass", "YAAD", "Optimisers", "ProximalAlgorithms", "DataFrameMacros", "PSIS", "RadonKA", "IntervalLinearAlgebra", "AMGCLWrap", "DictArrays", "RuleMiner", "EasyVega", "CometLogger", "ImageBinarization", "PyArrow", "MatrixFactorizations", "UnitfulLinearAlgebra", "LinearElasticityBase", "EnergyExpressions", "Sixel", "YFinance", "PyCallJLD2", "Rocketeer", "Presentation", "Kronecker", "DoubleFloats", "DedekindCutArithmetic", "FinancialToolbox", "FHIRClientXML", "ImplicitPlots", "TensorCrossInterpolation", "MonotonicSplines", "EcRequests", "SparseMatrixColorings", "Quante", "AtiyahBott", "Discord", "ExampleJuggler", "XmlStructWriter", "IBMQClient", "SwarmAgents", "ForwardDiffPullbacks", "CCBlade", "TransformVariables", "DataToolkit", "AngularMomentumAlgebra", "Polylabel", "Tensorial", "FunctionTabulations", "FoldRNA", "GeoInterfaceMakie", "Tensors", "BibParser", "ConstrainedShortestPaths", "ScHoLP", "TimeseriesFeatures", "Transducers", "RationalFunctionApproximation", "ConstitutiveModels", "BrillouinZoneMeshes", "Bibliography", "StandardizedRestrictedBoltzmannMachines", "DataFramesMeta", "NbodyGradient", "JMPReader", "RegionGrids", "ToyPublicKeys", "FiberNlse", "IsingModels", "AutoregressiveModels", "MvNormalCalibration", "MathTeXEngine", "FunctionFusion", "SparseMatricesCOO", "SoleBase", "BinaryBuilderAuditor", "InferenceObjects", "DensityEstimationDatasets", "LandauDistribution", "AssignTaxonomy", "DiscreteChoiceCalculations", "NomnomlJS", "IsotopeTableDF", "QuantumGradientGenerators", "CenteredRBMs", "DataToolkitCommon", "AdvRBMs", "TestTools", "TreeTools", "ImplicitGlobalGrid", "ChunkedCSV", "GaloisFields", "SliceMap", "Herb", "TrustRegionMethods", "WaterLily", "AcousticMetrics", "QuadraticFormsMGHyp", "PolyesterForwardDiff", "Experimenter", "Seaborn", "NCDatasets", "HePPCAT", "CCDReduction", "DistributionFits", "Ephemerides", "BundleAdjustmentModels", "SMLMMetrics", "HopTB", "GrafCSV", "LiiBRA", "SLEEFMath", "FlashAttentionWrapper", "BipolarSphericalHarmonics", "RangeEnclosures", "ConvolutionalOperatorLearning", "GPLikelihoods", "ImageSmooth", "GraphPPL", "JuDGE", "ExtendableSparse", "TensorGames", "UnifiedPseudopotentialFormat", "FactorRotations", "NeutralLandscapes", "SymFEL", "GreenFunc", "CatBoost", "SpectralResampling", "FastDMTransform", "UnicodePlots", "OnlinePortfolioAnalytics", "BPGates", "FastHistograms", "MarsagliaDiscreteSamplers", "SDeMo", "BoxLeastSquares", "LibSndFile", "MathJaxRenderer", "GraphDynamics", "NonlinearSolveBase", "Ipaper", "TriangularSolve", "FrechetDist", "DelayEmbeddings", "AISCSteel", "RecurrenceAnalysis", "CumulantsUpdates", "MutualInformationImageRegistration", "PlutoTeachingTools", "Bukdu", "BayesianLinearRegressors", "BOPTestAPI", "VectorizedStatistics", "MonteCarloSummary", "FinanceCore", "ReinforcementLearningTrajectories", "LaserTypes", "TopologicalNumbers", "PlutoPlotly", "MetidaBase", "SpatialEcology", "SparseExtra", "DataInterpolations", "CumulantsFeatures", "LightBSON", "ConstraintDomains", "PlutoStyles", "MAGEMin_C", "SciMLJacobianOperators", "ExplainableAI", "DifferentiableExpectations", "Octavian", "NLLSsolver", "BaytesMCMC", "TropicalGEMM", "GIFImages", "ImageFiltering", "SunAsAStar", "ProbabilityBoundsAnalysis", "PottsGumbelRBMLayers", "BlockDecomposition", "Photodynamics", "StaticWebPages", "ObjectPools", "SmoQyDEAC", "AcceleratedKernels", "CSDP", "GaussianMixtureRegressions", "LabelledArrays", "ComplexMixtures", "CategoricalMonteCarlo", "BifrostTools", "CircleFit", "PlantRayTracer", "SpheriCart", "JupyterPlutoConverter", "MetidaStats", "QPGreen", "SPECTrecon", "LinearMixingModels", "Gaius", "Powerful", "ShiftedProximalOperators", "MultiScaleTreeGraph", "LibRaw", "ElectronGas", "ImplicitBVH", "MetidaNCA", "AbstractGPs", "AbstractLogic", "AutomotiveSimulator", "AlgebraicMultigrid", "InducingPoints", "RegressionTables", "TemporalGPs", "FlowWorkspace", "RandomWalkBVP", "Vlasiator", "VisualGeometryOptimization", "PRASCapacityCredits", "StrategicGames", "BitSAD", "STREAMBenchmark", "Mice", "Stheno", "AbsSmoothFrankWolfe", "SlidingDistancesBase", "ElementaryFluxModes", "FastGeoProjections", "VectorizedReduction", "RandomFeatureMaps", "BoundaryValueProblems", "PowerModelsAnnex", "UMAP", "GaussianMixtures", "MLUtils", "DistributionMeasures", "ComplexityMeasures", "WaterModels", "BigO", "CoordRefSystems", "ExtendableGrids", "ChowLiuTrees", "EnergyModelsHeat", "Determinantal", "JetReconstruction", "ComputerVisionMetrics", "FastTransforms", "EnergyModelsRenewableProducers", "MatrixProfile", "FileTrees", "NeumannKelvin", "PlmDCA", "FractalDimensions", "SlottedRandomAccess", "GEMPIC", "Santiago", "GeoParams", "InventoryManagement", "SpecTools", "VLBILikelihoods", "InfiniteOpt", "CTBase", "QHull", "HOODESolver", "SimplexGridFactory", "PCquery", "DifferentiableFrankWolfe", "TimeseriesSurrogates", "GAP", "MicroscopePSFs", "ZarrDatasets", "SymbolicUtils", "FrankWolfe", "SeeToDee", "KSVD", "JobSchedulers", "PolynomialAmoebas", "SkyDomes", "RegressionDynamicCausalModeling", "NASAPrecipitation", "SurveyDataWeighting", "RadiationPatterns", "Mango", "TransmuteDims", "TransferFunctions", "ClimateERA", "GraphSignals", "LocalPoly", "XCALibre", "EwaldSummations", "FourierTools", "ImageIO", "ModiaResult", "OptimizationOptimisers", "SignalTemporalLogic", "TensorCast", "GEOTRACES", "MTH229", "TcpInstruments", "GasPowerModels", "LowLevelParticleFilters", "DifferentiableMetabolism", "DocstringAsImage", "ReferenceFiniteElements", "LifeContingencies", "FeynmanDiagram", "MarkovChainHammer", "ApproxFunFourier", "OptimizationPRIMA", "KiteUtils", "NURBS", "SNOW", "Baytes", "FinEtoolsHeatDiff", "PossibilisticArithmetic", "FinEtoolsAcoustics", "DAEProblemLibrary", "SliceSampling", "DASSL", "BVProblemLibrary", "DDEProblemLibrary", "AbstractCosmologicalEmulators", "SodShockTube", "TableTransforms", "ModiaBase", "InfrastructureSystems", "FinEtools", "FinEtoolsMultithreading", "CurvilinearGrids", "Sensemakr", "ChebParticleMesh", "TensorKitSectors", "DASKR", "ThermodynamicIntegration", "SignalAlignment", "Thermochron", "RedClust", "DiffEqBase", "KomaMRICore", "ParticleFilters", "GraphsOptim", "SphericalScattering", "DACE", "ArDCA", "Jadex", "DynamicHMC", "ProfileView", "SimpleDiffEq", "MotivationalQuotes", "AtomsCalculatorsUtilities", "FinEtoolsFlexStructures", "EmpiricalPotentials", "TensorOperationsTBLIS", "ImageQualityIndexes", "SimulationLogs", "GridVisualize", "HuggingFaceDatasets", "FinEtoolsDeforLinear", "SIMIlluminationPatterns", "MRFingerprintingRecon", "EtherSPH", "DecisionMakingPolicies", "JOLI", "Polynomials4ML", "Gadfly", "MultiData", "SMLMFrameConnection", "GLFixedEffectModels", "FiniteElementContainers", "OrdinaryDiffEqQPRK", "OrdinaryDiffEqHighOrderRK", "SciMLNLSolve", "NonconvexCore", "OrdinaryDiffEqFeagin", "OrdinaryDiffEqLowOrderRK", "OrdinaryDiffEqStabilizedRK", "IRKGaussLegendre", "CalciumScoring", "Zauner", "LSODA", "OrdinaryDiffEqNordsieck", "OrdinaryDiffEqRKN", "ODE", "OrdinaryDiffEqAdamsBashforthMoulton", "NonconvexSemidefinite", "MolSimToolkit", "PhysicalMeshes", "NonconvexSearch", "SUNRepresentations", "OrdinaryDiffEqSSPRK", "VisualRegressionTests", "OptimizationNLopt", "SimSearchManifoldLearning", "NonconvexNLopt", "PowerSystemCaseBuilder", "SpeedMapping", "SpinGlassTensors", "EclipsingBinaryStars", "DrugInteractions", "VMRobotControl", "PredefinedDynamicalSystems", "PassiveTracerFlows", "MetidaFreq", "JUDI", "LowRankLayers", "DiffEqPhysics", "LinearRegressionKit", "ParticleInCell", "MetidaNLopt", "PsychometricsBazaarBase", "FusibleBroadcasts", "ParallelAnalysis", "ModiaPlot_PyPlot", "TransitionsInTimeseries", "NonconvexMetaheuristics", "AreaInterpolation", "FieldTracer", "OrdinaryDiffEqSymplecticRK", "ODEInterfaceDiffEq", "GMT", "SignalTablesInterface_PyPlot", "TaylorModels", "MolecularGraphKernels", "GraphNets", "TaylorIntegration", "PhysicalFFT", "Sundials", "Thebes", "DiscoDiff", "ConstraintExplorer", "SciPyDiffEq", "VLBIImagePriors", "CaratheodoryFejerApprox", "PlasmaEquilibriumToolkit", "CategoryData", "Metida", "RecurrentLayers", "NonconvexMMA", "Wflow", "MPIMeasurements", "TaylorInversion", "InvariantPointAttention", "RobustNeuralNetworks", "CoordGridTransforms", "SimpleBoundaryValueDiffEq", "NonconvexNOMAD", "Associations", "IESopt", "StartUpDG", "ConjugateComputationVI", "NonconvexPercival", "Jabalizer", "PhysicalFDM", "UnderwaterAcoustics", "FMIExport", "TulipaEnergyModel", "MCMCDebugging", "PALEOboxes", "ReversePropagation", "TruncatedMVN", "EconomicScenarioGenerators", "FractionalDiffEq", "GaussianVariationalInference", "IntervalConstraintProgramming", "SteadyWaves", "FMIBuild", "PiecewiseDeterministicMarkovProcesses", "ClimaTimeSteppers", "MCMCChains", "OrdinaryDiffEqExtrapolation", "BoundaryValueDiffEqCore", "FMIBase", "DynACof", "HomalgProject", "CellSegmentation", "IterativeLQR", "SphericalFunctions", "TimeSeriesClassification", "UnROOT", "GeoTables", "NDTensors", "Yields", "GslibIO", "MeshIntegrals", "PyBraket", "PolaronMobility", "RelevancePropagation", "BoundaryValueDiffEqFIRK", "MarginalLogDensities", "JetPack", "BoundaryValueDiffEqMIRKN", "TaylorDiff", "OptimizationOptimJL", "TuringBenchmarking", "ArviZPythonPlots", "Qaintmodels", "SignalPlots", "WordCloud", "ParametricMCPs", "ElectronLiquid", "Gabs", "YaoPlots", "RHEOS", "DiffFusion", "ONSAS", "VectorSpaceDarkMatter", "EnergySamplers", "PALEOsediment", "PSSFSS", "GridapTopOpt", "DiffEqFinancial", "Unfolding", "ClimaDiagnostics", "ExponentialFamilyManifolds", "QuantitativeSusceptibilityMappingTGV", "ManifoldDiffEq", "DeconvOptim", "ReinforcementLearning", "JsonGrinder", "OrdinaryDiffEqFIRK", "BoundaryValueDiffEqAscher", "DerivableFunctions", "MaximumEntropyMomentClosures", "DIVAnd_HFRadar", "Bcube", "OptimizationFlux", "BcubeVTK", "SymbolicControlSystems", "DifferentiableTrajectoryOptimization", "EquationsSolver", "FastMPOContractions", "ComputerAdaptiveTesting", "EvoLinear", "DistributedStwdLDA", "DrillHoles", "GeneralizedMonteCarlo", "BcubeCGNS", "Dolo", "ExpressionTreeForge", "TensorTrains", "ROMEO", "Swalbe", "EvoDynamics", "SuperfluidRotSpec", "FSimZoo", "DirectTrajectoryOptimization", "SwitchOnSafety", "UlamMethod", "OrdinaryDiffEqBDF", "OptimizationSpeedMapping", "GMMParameterEstimation", "ClosedLoopReachability", "SampleChainsDynamicHMC", "GlobalSensitivity", "ReinforcementLearningFarm", "MriResearchTools", "PowerPlots", "SpinGlassNetworks", "ConstrainedDynamicsVis", "SymbolicAnalysis", "GeoStatsBase", "MGVI", "ProbabilisticCircuits", "DIVAnd", "ReinforcementLearningCore", "GlobalApproximationValueIteration", "PartiallySeparableNLPModels", "PETLION", "GeoEnergyIO", "Walrus", "SurrogatesRandomForest", "ActuaryUtilities", "GenericCharacterTables", "VortexDistributions", "WaveOpticsPropagation", "POMDPModelChecking", "ITensorVisualizationBase", "RealPolyhedralHomotopy", "NeXLMatrixCorrection", "Jutul", "PawsomeTracker", "SourceCodeMcCormick", "QSFit", "StellaratorOptimization", "MetidaBioeq", "ExponentialFamilyProjection", "CountriesBorders", "MatterPower", "MicrobeAgents", "GridapGmsh", "Yao", "FSimROS", "GeoStatsModels", "ParametrisedConvexApproximators", "DynamicMovementPrimitives", "JointEnergyModels", "SocialSamplingTheory", "ImageQuilting", "ApproxMasterEqs", "TropicalYao", "StellaratorOptimizationMetrics", "GeoStatsFunctions", "RegressionDiscontinuity", "BattMo", "FluxTraining", "Consensus", "VMEC", "TropicalNN", "ReinforcementLearningZoo", "GeoGrids", "Phonetics", "YaoSubspaceArrayReg", "ITensorUnicodePlots", "DeepQLearning", "RvSpectML", "GeoEstimation", "Omniscape", "Circuitscape", "Attractors", "BloqadeExpr", "FNCFunctions", "Gogeta", "BloqadeDormandPrince", "SchwarzChristoffel", "LogicCircuits", "ObjectDetector", "FixedPointToolkit", "MultiStateSystems", "NonconvexJuniper", "NBodySimulator", "Images", "MomentMatching", "FastBEAST", "Eikonal", "MaxEntropyGraphs", "NonconvexIpopt", "Globtim", "ColorSchemeTools", "Tasmanian", "GeneralizedSDistributions", "ODEHybrid", "GameTheory", "ImageTracking", "StatisticalRethinking", "ImageFeatures", "ChaoticEncryption", "MimiRFFSPs", "GeoStatsTransforms", "FractionalSystems", "GeoStatsSolvers", "ImageProjectiveGeometry", "NonconvexPavito", "BloqadeMIS", "CompressedBeliefMDPs", "EMpht", "GeometricalOptics", "HierarchicalEOM", "BundlerIO", "PALEOmodel", "SpaSM", "BLASBenchmarksCPU", "ImageUtils", "BaseModelica", "DecomposingPolynomialSystems", "NonconvexBayesian", "BloodFlowTrixi", "StarFormationHistories", "NonlinearSolveHomotopyContinuation", "AffineMotions", "RvLineList", "MatrixProductBP", "ECCO", "PerceptualColourMaps", "ClimateTools", "ParameterizedFunctions", "ImageHashes", "ColBERT", "vSmartMOM", "CirculatorySystemModels", "IndependentComponentAnalysis", "PyGDatasets", "Fronts", "DynamicalSystems", "EverySingleStreet", "TwoStageOptimalControl", "ClimatePlots", "RPRMakie", "JumpProblemLibrary", "ODEProblemLibrary", "FinEtoolsVibInFluids", "Colocalization", "AstrodynamicalModels", "Scheduling", "StochasticDelayDiffEq", "TensorQEC", "SDEProblemLibrary", "BloqadeWaveforms", "ViscousFlow", "LocalAnisotropies", "LiquidElectrolytes", "DiffusionGarnet", "GraphMakie", "ConceptualClimateModels", "FrequencySweep", "PairPlots", "ActiveInference", "SymbolicNumericIntegration", "CellMLToolkit", "SIAN", "GpABC", "TagPOMDPProblem", "ChargeTransport", "MakieThemes", "OptimizationMOI", "Makie", "GeoStats", "AdditiveCellCom", "PRONTO", "GeometricFlux", "PubChem", "FaceDetection", "MNPDynamics", "SpiDy", "AdaOPS", "BloqadeQMC", "HierarchicalGaussianFiltering", "ClapeyronHANNA", "BloqadeKrylov", "Mellan", "DataDrivenDMD", "CitableImage", "NetworkJumpProcesses", "SolverBenchmark", "DataDrivenSparse", "Bactos", "DifferentiableBackwardEuler", "NamedTrajectories", "Fable", "Petri", "CitablePhysicalText", "WGPUgfx", "ForestMensuration", "GeoMakie", "BaytesInference", "XGPaint", "Vahana", "SurfaceReactions", "LowRankIntegrators", "ElectrochemicalKinetics", "StirredReactor", "Pesto", "RealTimeAudioDiffEq", "MonolithicFEMVLFS", "EditorsRepo", "ChainPlots", "BatchReactor", "SurfaceCoverage", "Supernovae", "PlugFlowReactor", "OSMMakie", "MeanFieldToolkit", "SubsidenceChron", "Chron", "TidierPlots", "EasyABM", "Tapestree", "TaijaData", "HetaSimulator", "OpenQuantumSystems", "PortfolioAnalytics", "JointSurvivalModels", "NuclearToolkit", "ModiaPlot_CairoMakie", "RandomFeatures", "AstroIC", "MINDFulMakie", "FSimPlots", "SpatiallySymmetricTensors", "Jchemo", "WaveletsExt", "SymbolicInference", "MultiscaleGraphSignalTransforms", "Psychrometrics", "InternalFluidFlow", "PDMPFlux", "QuantumCollocationCore", "LineIntegralConvolution", "NVMagnetometer", "SimulatedNeuralMoments", "MAGEMinApp", "QuanEstimationBase", "SequentialSamplingModels", "StateSpacePartitions", "GalacticPotentials", "VeriQuEST", "Collide", "QuantumOptics", "MixedComplementarityProblems", "GeneticsMakie", "HiQGA", "MathepiaModels", "BaryPlots", "CollectiveSpins", "PhyloClustering", "BoxCox", "WaveguideQED", "SymBoltz", "GeoIO", "TemporalNetworks", "Microstructure", "IonSim", "CropRootBox", "NighttimeLights", "FourLeafMLE", "DiskArrayEngine", "HmtGutenberg", "SBMLToolkitTestSuite", "HmtArchive", "QuantumDynamics", "Biofilm", "Coalescent", "SMLMSim", "Population", "CalibrateEmulateSample", "BloqadeGates", "RigorousInvariantMeasures", "TulipaPlots", "SwissVAMyKnife", "TaijaPlotting", "CarnotCycles", "MiseEnPage", "NeuroAnalysis", "SideKicks", "VlasovMethods", "Chamber", "GIRFReco"], vs = ":release-1.11")`

KristofferC commented:
Will start merging these a bit more frequently since people can now "subscribe" to the release branch using the `1.12-nightly` channel in juliaup.

nanosoldier commented:
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/57258#issuecomment-2655848411) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/ebef610_vs_d63aded/report.html) is available.

<details><summary>Report summary</summary>

## ❗ Packages that crashed

**84 packages crashed only on the current version.**

 - The process was aborted: 14 packages
 - Invalid LLVM IR was generated: 1 packages
 - An internal error was encountered: 62 packages
 - An unreachable instruction was executed: 1 packages
 - GC corruption was detected: 1 packages
 - A segmentation fault happened: 5 packages


## ✖ Packages that failed

**892 packages failed only on the current version.**

 - Package has syntax issues: 11 packages
 - Package fails to precompile: 510 packages
 - Illegal method overwrites during precompilation: 17 packages
 - Package has test failures: 68 packages
 - Package tests unexpectedly errored: 135 packages
 - There were unidentified errors: 7 packages
 - Tests became inactive: 3 packages
 - Test duration exceeded the time limit: 139 packages
 - Test log exceeded the size limit: 2 packages

40 packages failed on the previous version too.

## ✔ Packages that passed tests

96 packages passed tests on the previous version too.



</details>
